src/widgets/dialogs/LogsPopup.cpp

```diff
@@ -0,0 +1,151 @@
+#include "widgets/helper/ChannelView.hpp"
+
+#include <QDateTime>
+#include <QMessageBox>
+#include <QVBoxLayout>
+
+namespace chatterino {
+
+LogsPopup::LogsPopup()
+{
+    this->initLayout();
+    this->resize(400, 600);
+}
+
+void LogsPopup::initLayout()
+{
+    {
+        QVBoxLayout *layout = new QVBoxLayout(this);
+        layout->setMargin(0);
+
+        {
+            this->channelView = new ChannelView(this);
+            layout->addWidget(this->channelView);
+        }
+
+        this->setLayout(layout);
+    }
+}
+
+void LogsPopup::setInfo(ChannelPtr channel, QString userName)
+{
+    this->channel_ = channel;
+    this->userName = userName;
+    this->setWindowTitle("Logs for " + this->userName + " in " + this->channel_->name +
+                         "'s history");
+    this->getLogviewerLogs();
+}
+
+void LogsPopup::setupView(std::vector<MessagePtr> messages)
+{
+    ChannelPtr logsChannel(new Channel("logs", Channel::Misc));
+    /*
+                MessagePtr message(new Message);
+                message->addElement(new TimestampElement(this->rustleTime));
+                message->addElement(
+                    new TextElement(this->userName, MessageElement::Username,
+       MessageColor::System)); message->addElement( new TextElement(output[line],
+       MessageElement::Text, MessageColor::Text)); messages.push_back(message);*/
+    logsChannel->addMessagesAtStart(messages);
+    this->channelView->setChannel(logsChannel);
+}
+
+void LogsPopup::getLogviewerLogs()
+{
+    TwitchChannel *twitchChannel = dynamic_cast<TwitchChannel *>(this->channel_.get());
+    if (twitchChannel == nullptr) {
+        return;
+    }
+
+    QString channelName = twitchChannel->name;
+
+    QString url = QString("https://cbenni.com/api/logs/%1/?nick=%2&before=500")
+                      .arg(channelName, this->userName);
+
+    NetworkRequest req(url);
+    req.setCaller(QThread::currentThread());
+
+    req.onError([this](int errorCode) {
+        this->getOverrustleLogs();
+        return true;
+    });
+
+    req.getJSON([this, channelName](QJsonObject &data) {
+
+        std::vector<MessagePtr> messages;
+        ChannelPtr logsChannel(new Channel("logs", Channel::None));
+
+        QJsonValue before = data.value("before");
+
+        for (auto i : before.toArray()) {
+            auto messageObject = i.toObject();
+            QString message = messageObject.value("text").toString();
+
+            // Hacky way to fix the timestamp
+            message.insert(1, "historical=1;");
+            message.insert(1, QString("tmi-sent-ts=%10000;").arg(messageObject["time"].toInt()));
+
+            MessageParseArgs args;
+            auto ircMessage = Communi::IrcMessage::fromData(message.toUtf8(), nullptr);
+            auto privMsg = static_cast<Communi::IrcPrivateMessage *>(ircMessage);
+            TwitchMessageBuilder builder(logsChannel.get(), privMsg, args);
+            if (!builder.isIgnored()) {
```

Comment:
Don't check if the message is ignored since the user wants to see all the messages when he is looking at the logs of a person. (e.g. if they are a mod)